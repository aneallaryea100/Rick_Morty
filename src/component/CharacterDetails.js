import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { useSelector, useDispatch } from 'react-redux';
import { getCharacterData } from '../redux/characters';

function CharacterDetails(props) {
  const { buttonId } = props;
  const dispatch = useDispatch();
  const characters = useSelector((state) => state.characterSlice);
  const checkCharacterId = characters.filter(
    (item) => item.id === buttonId,
  );

  useEffect(() => {
    dispatch(getCharacterData()).unwrap();
  }, []);
  return (
    <>
      {checkCharacterId.map((character) => (
        <div key={character.id} className="containerDetail">
          <div className="imageCharacter">
            <img
              alt={character.name}
              src={character.image}
              className="mainImage"
            />
          </div>
          <div className="listDetail">
            <li>
              Name:
              {character.name}
            </li>
            <li>
              Status:
              {character.status}
            </li>
            <li>
              Gender:
              {character.gender}
            </li>
            <li>
              Origin:
              {character.origin}
            </li>
            <li>
              Location:
              {character.location}
            </li>
            <Link to="/">
              <button className="backBtn" type="button">
                Back
              </button>
            </Link>
          </div>
        </div>
      ))}
    </>
  );
}

export default CharacterDetails;

CharacterDetails.propTypes = {
  buttonId: PropTypes.number.isRequired,
};
