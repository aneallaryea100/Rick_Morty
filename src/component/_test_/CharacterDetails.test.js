import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import CharacterDetails from '../CharacterDetails';
import store from '../../redux/configureStore';

describe('CharacterDetails test', () => {
  test('snapshot', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <CharacterDetails />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
