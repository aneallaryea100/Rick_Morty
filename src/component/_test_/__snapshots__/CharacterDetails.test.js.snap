// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetails test snapshot 1`] = `null`;
